CMakeLists.txt

```
@@ -185,3 +185,16 @@ function(mbed_generate_executable target)
     mbed_generate_bin_hex(${target})
     mbed_generate_map_file(${target})
 endfunction()
+
+# Ninja requires to be forced for response files
+if ("${CMAKE_GENERATOR}" MATCHES "Ninja")
+    # known issue ARMClang and Ninja with response files for windows
+    # https://gitlab.kitware.com/cmake/cmake/-/issues/21093
+    set(ninja_response_enable 1)
+    if (CMAKE_HOST_SYSTEM_NAME MATCHES "Windows" AND CMAKE_CXX_COMPILER_ID MATCHES "ARMClang")
+        set(ninja_response_enable 0)
+    endif()
+    if(ninja_response_enable)
```

Can't the following be used?
```suggestion
    if(NOT (CMAKE_HOST_SYSTEM_NAME MATCHES "Windows" AND CMAKE_CXX_COMPILER_ID MATCHES "ARMClang"))
```